src/fft.jl

```diff
@@ -88,9 +88,8 @@ function FFTGrid(fft_size::Tuple{Int, Int, Int}, unit_cell_volume::T,
     fft_normalization  = sqrt(unit_cell_volume) / length(ipFFT)
 
     VT = value_type(T)
-    r_vectors = [(Vec3{VT}(idx.I) .- (1, 1, 1)) ./ VT.(fft_size)
-                 for idx in CartesianIndices(fft_size)]
-    r_vectors = to_device(arch, r_vectors)
+    fac = 1 ./ VT.(fft_size)
+    r_vectors = map(idx -> Vec3{VT}(fac .* (idx.I .- (1, 1, 1))), CartesianIndices(fft_size))
```

Comment:
I may be blind, but why does this now do computation on the GPU ?

Reply:
You are right, it does not. I have been blindsided because it is so much faster, and because removing the `to_device()` function does not break the GPU tests, nor any example I ran.

Browsing the code, it is not clear to me whether `r_vectors` should be a GPU array at all or not, actually

Reply:
Maybe it does not need to be. For consistency it is, however, very strange if `r_vectors(basis)` returns a CPU array but `G_vectors(basis)` does return a GPU array. Could the solution be to do the computation on the CPU and then only transfer it to the GPU thereafter ?

Reply:
Given the speed of the operation, this is very reasonable. And we get peace of mind and consistency.